### PR TITLE
CIV-4306 Hearing Fee Payment

### PR DESCRIPTION
**Pull request by miguelMolina3691**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by sankhajuria — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleService.java
Comment (miguelMolina3691) [inline src/main/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleService.java:102]: reset before merge

File: src/test/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleServiceTest.java
Comment (miguelMolina3691) [inline src/test/java/uk/gov/hmcts/reform/civil/launchdarkly/FeatureToggleServiceTest.java:209]: Do not merge to master

**Review by m-meulendijks-v1 — DISMISSED** [review]

**Review by maciejmodzelewski — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandler.java
Comment (filippo-possenti-version1) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandler.java]: When I see try/catch it always sets me off. It might be perfectly fine, but I need to ask, especially if there is no dedicated comment.

- Why is it expected that the call to getHearingFeeDataByTotalClaimAmount could throw an exception?
- What are the implications of a failure in calling getHearingFeeDataByTotalClaimAmount?

Comment (miguelMolina3691) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandler.java]: it was only a check for testing, I have removed it

Comment (filippo-possenti-version1) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandler.java]: Ah, ok.

**Review by filippo-possenti-version1 — APPROVED** [review]
Approved.

Next time in new tests put the `// Given`, `// When`, `// Then` comments for clarity.

**Review by maciejmodzelewski — DISMISSED** [review]

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by filippo-possenti-version1 — DISMISSED** [review]

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Comment by filippo-possenti-version1** [comment]
The build is green and no overlapping changes compared to master are present.
In the interest of speed I'll be merging this even though the branch needs updating.

**Pull request merged**
